R/mail.R

```diff
@@ -0,0 +1,109 @@
+#' @param reply_datetime Date, time in the following format: 2023-02-11T09:30:00  Be sure to add the capital T to separate date and time
+#' @param timezone  The abbreviation of desired timezone eg: HST
+#' @param reply_subject subject of the email as a string
+#' @param reply_body the body of the email as a string
+#' @return the content of the fake mbox you created
+#' @export
+create_fake_mbox_replies <- function(mlist, reply_from_author, reply_from_email, reply_to_author, reply_to_email, reply_cc_author, reply_cc_email, reply_datetime, timezone, reply_subject, reply_body) {
+
+# format the date correctly
+  cdate <- format(as.POSIXct(reply_datetime, format = "%Y-%m-%dT%H:%M:%S"), "%a, %e %b %Y %H:%M:%S ")
+
+ reply_from_full_info <- paste0(reply_from_author, " <", reply_from_email, "> ")
+ reply_to_full_info <- paste0(reply_to_author, " <", reply_to_email, "> ")
+ reply_cc_full_info <- paste0(reply_cc_author, " <", reply_cc_email, "> ")
+
+
+  mbox_content <- paste0(
+    "From MAILER-DAEMON Thu Jul 18 13:48:48 2013 ",
+    "\nPath: example.com!not-for-mail ",
+    "\nFrom: ", reply_from_full_info,
+    "\nNewsgroups: gmane. ", mlist,
+    "\nSubject: ", reply_subject,
+    "\nDate: ", cdate, timezone,
+    "\nApproved: auto ",
+    "\nMessage-ID: <", length(reply_body),
+    "\nNNTP-Posting-Host: example.com",
+    "\nMime-Version: 1.0 ",
+    "\nContent-Type: text/plain; charset=us-ascii; format=flowed ",
+    "\nContent-Transfer-Encoding: 7bit ",
+    "\nX-Complaints-To: complaints@example.com ",
+    "\nNNTP-Posting-Date: ", cdate, timezone,
+    "\nUser-Agent: Mozilla/5.0 (X11; U; Linux i686; en-US; rv:0.9.8) Gecko/20020205 ",
+    "\nX-Accept-Language: en-us ",
+    "\nOriginal-To: ", reply_to_full_info, " ", reply_cc_full_info,
+    "\nPrecedence: bulk",
+    "\nX-Mailing-List: ", paste0(mlist, "@example.com "),
+    "\n\n", reply_body
+  )
+
+  return(mbox_content)
+}
+
+
+#' Takes in mbox replies and creates a .mbox file
+#'
+#' Takes a list of mbox replies generated with create_fake_mbox_replies function and compiles them all into a single
+#' fake .mbox file
+#'
+#' @param folder_path  Folder path for the .mbox file being created. Defaulted at /tmp
+#' @param folder_name Name of the folder that will store the .mbox file
+#' @param replies A list of replies that have been created with create_fake_mbox_replies() function that will be added to .mbox file
+#' @return the path of the .mbox file that was created, now that each reply has been added
+#' @export
+create_mbox_from_replies <- function(folder_path = "/tmp", folder_name, replies) {
+  # Create an mbox file from a list of replies
+
+  # Create a unique filename for the mbox file
+  mbox_filepath <- file.path(folder_path, paste0(folder_name, ".mbox"))
+
+  # Open the mbox file for writing
+  mbox_file <- file(mbox_filepath, "w")
+
+  # Iterate through each reply and write it to the mbox file
+  for (reply in replies) {
+    cat(reply, "\n\n", file = mbox_file)
```

Comment:
We should avoid spreading internal functionality already defined in Kaiaulu in other functions (makes it harder to modify the functionality if its internals are everywhere). In addition, this is performing multiple i/o operations. We can concatenate the strings on memory, and then save them to disk in one i/o operation.

Try this:

mbox_body <- stringi::stri_c(replies,collapse = "\n\n")
io_make_file(mbox_filepath,mbox_body)